File: bentley-ottmann/src/event.cpp
#include "event.h"

event::event(point _point, segment _segment, int _type) :
	p(_point), s(_segment), type(_type) {
}

double event::get_x_coord() const {
	return this->p.get_x_coord();
}

double event::get_y_coord() const {
	return this->p.get_y_coord();
}

int event::get_type() const {
	return this->type;
}

segment event::get_segment() const {
	return this->s;
}

point event::get_point() const {
	return this->p;
}



event::~event() {
	
}

File: bentley-ottmann/src/bentley_ottmann/Event.java
package bentley_ottmann;

import java.util.*;

/**
 * Created by valen_000 on 14. 5. 2017.
 */

public class Event {

    private Point point;
    private ArrayList<Segment> segments;
    private double value;
    private int type;

    Event(Point p, Segment s, int type) {
        this.point = p;
        this.segments = new ArrayList<>(Arrays.asList(s));
        this.value = p.get_x_coord();
        this.type = type;
    }

    Event(Point p, ArrayList<Segment> s, int type) {
        this.point = p;
        this.segments = s;
        this.value = p.get_x_coord();
        this.type = type;
    }

    public void add_point(Point p) {
        this.point = p;
    }

    public Point get_point() {
        return this.point;
    }

    public void add_segment(Segment s) {
        this.segments.add(s);
    }

    public ArrayList<Segment> get_segments() {
        return this.segments;
    }

    public void set_type(int type) {
        this.type = type;
    }

    public int get_type() {
        return this.type;
    }

    public void set_value(double value) {
        this.value = value;
    }

    public double get_value() {
        return this.value;
    }

}
